Watchman.DomainModel/Messages/Commands/Handlers/AddMessagesCommandHandler.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Messages.Commands.Handlers
+{
+    public class AddMessagesCommandHandler : ICommandHandler<AddMessagesCommand>
+    {
+        private List<string> _cashedExistingMessagesHashes;
+        private ulong _cashedChannelId;
+
+        private readonly ISessionFactory _sessionFactory;
+
+        public AddMessagesCommandHandler(ISessionFactory sessionFactory)
+        {
+            _sessionFactory = sessionFactory;
+        }
+
+        public async Task HandleAsync(AddMessagesCommand command)
+        {
+            var newMessages = await Task.Run(() => GetOnlyNewMessages(command.Messages, command.ChannelId));
+            using var session = _sessionFactory.Create();
+            foreach (var message in newMessages)
+            {
+                session.Add(message);
+            };
+        }
+
+        private IEnumerable<Message> GetOnlyNewMessages(IEnumerable<Message> messages, ulong channelId)
+        {
+            using var session = _sessionFactory.Create();
+            var existingHashes = GetExistingHashes(channelId);
+
+            return messages.Where(x =>
+            {
+                var isMessageNew = existingHashes.BinarySearch(GetHash(x)) < 0;
+                return isMessageNew;
+            });
+        }
+
+        private List<string> GetExistingHashes(ulong channelId)
+        {
+            if (_cashedChannelId == channelId)
+            {
+                return _cashedExistingMessagesHashes;
+            }
+
+            using var session = _sessionFactory.Create();
+            var channelMessagesHashes = session.Get<Message>()
```

Comment:
myśle że lepszą opcją będzie trzymanie tego w bazie 

hashowanie wszystkich pobranych danych za każdym razem nie brzmi jak coś dobrego 
może osobna kolekcja w mongo to ok rozwiązanie 
albo osobne pole do każdej wiadomości w bazie
w którym będzie hash tej wiadomości

Reply:
plus trzeba oznaczać że chodzi akurat o hash MD5